scipy/optimize/_remove_redundancy.py

```diff
@@ -185,6 +185,8 @@ def _remove_redundancy_dense(A, rhs):
     # mistakes and ambiguities - which is strange, because the rest of
     # Andersen's papers are quite good.)
 
+    lu = np.eye(m, order='F'), np.arange(m)  # LU for initial basis is trivial
+    perm_r = lu[1]
```

Comment:
Originally, the algorithm would hit an exception in the `try` block because `lu` was `None`, then `lu` would get created in the `except` by `lu = scipy.linalg.lu_factor(B)`. But `B` is initially the identity matrix, so it is easy (and faster) to define it manually.